src/radix_connect/app_link_parser.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+use url::form_urlencoded;
+use url::Url;
+
+use super::{DappRequest, LinkRequest, MobileConnectRequest};
+
+#[uniffi::export]
+pub fn parse_request_url(url: String) -> Result<MobileConnectRequest> {
+    let connect_url = Url::parse(CONNECT_URL).unwrap();
+    match Url::parse(url.as_str()) {
```

Comment:
to reduce nesting I think you can this:

```rust
let parsed_url = Url::parse(url.as_str())
        .map_err(|_| CommonError::RadixConnectMobileInvalidRequestUrl)?;
```
this will throw immediately if the error is encountered. with this you can remove the whole `match`

Reply:
ha I should have use "Add single comment", my suggestion is better than this one :) even more generalized :)